web/src/components/tasks/xray-registration/XrayRegistrationStatus.tsx

```diff
@@ -0,0 +1,218 @@
+
+const useStyles = makeStyles(() => {
+  return createStyles({
+    addressZipCodeField: {
+      "& input::-webkit-clear-button, & input::-webkit-outer-spin-button, & input::-webkit-inner-spin-button":
+        {
+          display: "none",
+        },
+    },
+  });
+});
+
+interface Props {
+  onSubmit: (facilityDetails: FacilityDetails) => void;
+  error: XraySearchError | undefined;
+  isLoading: boolean;
+}
+
+const Config = getMergedConfig();
+
+const XrayRegistrationErrorLookup: Record<XraySearchError, string> = {
+  NOT_FOUND: Config.xrayRegistrationTask.errorTextNotFound,
+  FIELDS_REQUIRED: Config.xrayRegistrationTask.errorTextFieldsRequired,
+  SEARCH_FAILED: Config.xrayRegistrationTask.errorTextSearchFailed,
+};
+
+export const XrayRegistrationStatus = (props: Props): ReactElement => {
+  const classes = useStyles();
+  const [formValues, setFormValues] = useState<FacilityDetails>({
+    businessName: "",
+    addressLine1: "",
+    addressLine2: "",
+    addressZipCode: "",
+  });
+  const { business } = useUserData();
+
+  useMountEffectWhenDefined(() => {
+    if (!business) return;
+
+    const nameAndAddress = business.xrayRegistrationData?.facilityDetails;
+
+    const hasNameAndAddress =
+      nameAndAddress?.businessName && nameAndAddress?.addressLine1 && nameAndAddress?.addressZipCode;
+
+    if (nameAndAddress && hasNameAndAddress) {
+      setFormValues(nameAndAddress);
+    } else if (business.formationData?.formationResponse?.success) {
+      setFormValues((prevValues) => {
+        return {
+          ...prevValues,
+          businessName: business.formationData.formationFormData.businessName,
+          addressLine1: business.formationData.formationFormData.addressLine1,
+          addressLine2: business.formationData.formationFormData.addressLine2 || "",
+          addressZipCode: business.formationData.formationFormData.addressZipCode,
+        };
+      });
+    } else {
+      setFormValues((prevValues) => {
+        return {
+          ...prevValues,
+          businessName: business.profileData?.businessName,
+          addressLine1: business.formationData?.formationFormData?.addressLine1,
+          addressLine2: business.formationData?.formationFormData?.addressLine2 || "",
+          addressZipCode: business.formationData?.formationFormData?.addressZipCode,
+        };
+      });
+    }
+  }, business);
+
+  const onSubmit = (event: FormEvent<HTMLFormElement>): void => {
+    event.preventDefault();
+    analytics.event.task_address_form.submit.submitted_address_form();
+    props.onSubmit(formValues);
+  };
+
+  const handleChangeForKey = (
+    key: keyof FacilityDetails
+  ): ((event: ChangeEvent<HTMLInputElement>) => void) => {
+    return (event: ChangeEvent<HTMLInputElement>): void => {
+      setFormValues((prevValues) => {
+        return {
+          ...prevValues,
+          [key]: event.target.value,
+        };
+      });
+    };
+  };
+
+  const getErrorAlert = (): ReactElement => {
```

Comment:
[dust] Should we be using `ReactNode` in favor of `ReactElement`, even though we know everything will be a single JSX element?